src/components/_button.scss

```diff
@@ -35,6 +35,10 @@
     @include sui-button--flat;
   }
 
+  @if $type == 'disabled' {
```

Comment:
Disabled isn't a type, it's a state. Anyway, it's not defined in the UI guidelines.